proxy-router/internal/blockchainapi/service.go

```diff
@@ -621,6 +619,26 @@ func (s *BlockchainService) GetSessions(ctx *gin.Context, user, provider common.
 	return mapSessions(ids, sessions, bids), nil
 }
 
+func (s *BlockchainService) GetSessionsIds(ctx *gin.Context, user, provider common.Address, offset *big.Int, limit uint8) ([]common.Hash, error) {
```

Comment:
no reason to inject whole gin request here in service. The parameters already parsed in controller

```suggestion
func (s *BlockchainService) GetSessionsIds(ctx context.Context, user, provider common.Address, offset *big.Int, limit uint8) ([]common.Hash, error) {
```